localstack/testing/pytest/fixtures.py

```diff
@@ -1316,22 +1317,33 @@ def make_response(body: dict, status_code: int = 200):
         }
 
 
+    def trim_headers(headers):
+        if not int(os.getenv("TRIM_X_HEADERS", 0)):
+            return headers
+        return {
+            key: value for key, value in headers.items()
+            if not (key.startswith("x-amzn") or key.startswith("x-forwarded-"))
```

Comment:
I am pretty impartial to what the default should be, I have been going back and forth. My thoughts for skipping by default are that it would make snapshots easier with less boiler plate transforms. As you mentioned that most snapshots do not usually include headers. Most care about the body. I also confirmed with httpbin and some headers, like `x-forwarded-` are also not echoed. And they also seem to use with most `x-amzn-` headers removed. But i will agree that it first lead to some confusing from me to see some headers not being returned! 

I can change the default to make it more transparent.

Agree with adding a snapshot tests for both versions. I will add those.